### ISSUE TITLE
[bug] "indexteam" models output whitespace only

### ISSUE THREAD

**Issue by lucyknada** [turboderp/exllamav2#546]
original model: https://huggingface.co/IndexTeam/Index-1.9B-Chat

the quant itself ran through just fine and the output can be found here: https://huggingface.co/EdgerunnersArchive/IndexTeam_Index-1.9B-Chat-exl2-6.0bpw

however when actually inferenced on latest dev-branch via tabbyAPI it just outputs empty characters.

the prompt template:

```
# The three delimiters are <unk>(token_id=0), reserved_0(token_id=3), reserved_1(token_id=4)
[<unk>]sytem_message[reserved_0]user_message[reserved_1]response
```

or

```
<unk>{{ if .System }}{{ .System }}{{ end }}reserved_0{{ .Prompt }}reserved_1
```

as extracted from here: https://huggingface.co/IndexTeam/Index-1.9B-Chat-GGUF/blob/main/OllamaModelFile

**Comment by turboderp** [comment]
It's most likely not quantizing correctly, and I assume you're getting a warning about an unsupported architecture. It falls back on Llama in that case, and the architecture seems to be similar enough to Llama to inference without crashing.

Just at a glance there's (at least) one breaking difference, being the `NormHead` layer. Looks like it's a regular head layer, but on the first forward pass it's normalized in-place:

```python
class NormHead(nn.Module):
    def __init__(self, hidden_size, vocab_size, bias=False):
        super().__init__()
        self.weight = nn.Parameter(torch.empty((vocab_size, hidden_size)))
        nn.init.kaiming_uniform_(self.weight, a=math.sqrt(5))
        self.first_flag = True

    def forward(self, hidden_states):
        if self.training:
            norm_weight = nn.functional.normalize(self.weight)
            self.first_flag = True
        elif self.first_flag:
            self.first_flag = False
            self.weight = nn.Parameter(nn.functional.normalize(self.weight))
            norm_weight = self.weight
        else:
            norm_weight = self.weight
        return nn.functional.linear(hidden_states, norm_weight)
```

I have no idea why they did it this way, but it definitely needs to be accounted for when converting the model.

I'll try to get to it soon, I guess.

**Comment by turboderp** [comment]
Alright, I added support and pushed the changes to the dev branch. It was the odd way in which the output layer weights are being normalized, and beyond that it's just a Llama model.

The chat template is weird, and I'm not sure what to make of it, but the model seems to work at any rate.
